lib/monetary-amount.spec.js

```diff
@@ -8,15 +8,137 @@ chai.use(sinonChai);
 const MonetaryAmount = require('./monetary-amount');
 const ethers = require('ethers');
 
+function cartesian (A, B) {
+    const C = [];
+
+    for (const a of A) {
+        for (const b of B) {
+            if (Array.isArray(b))
+                C.push([a, ...b]);
+            else
+                C.push([a, b]);
+        }
+    }
+
+    return C;
+}
+
+function bn (n) {
+    return ethers.utils.bigNumberify(n);
+}
+
+const legalAm = [
+    -1000, -1, 0, 1, 1000,
+    '-1000', '-1', '0', '1', '1000',
+    bn(-1000), bn(-1), bn(0), bn(1), bn(1000)
+];
+
+const illegalAm = [
+    undefined, null, {}, [],
+    '90f8bf6a479f320ead074411a4b0e7944ea8c9c1'
+];
+
+const legalCt = [
+    '90f8bf6a479f320ead074411a4b0e7944ea8c9c1',
+    '0000000000000000000000000000000000000000',
+    '0000000000000000000000000000000000000001',
+    '0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1',
+    '0x0000000000000000000000000000000000000000',
+    '0x0000000000000000000000000000000000000001'
+];
+
+const illegalCt = [
+    undefined, null, {}, [], -1, 0, 1,
+    '-1', '0', '1', '0x1', '0x0',
+    '90f8bf6a479f320ead074411a4b0e7944ea8c9cg',
+    '90f8bf6a479f320ead074411a4b0e7944ea8c9cg',
+    '090f8bf6a479f320ead074411a4b0e7944ea8c9c1',
+    'x90f8bf6a479f320ead074411a4b0e7944ea8c9c1',
+    '0X90f8bf6a479f320ead074411a4b0e7944ea8c9c1'
+];
+
+const legalId = [
+    undefined, 0, 1, 1000, '0x0', '0x1', '0xaf09', '0xAF09'
+];
+
+const illegalId = [
+    null, {}, [],-1000, -1, '-1000', '-1', 'x0', '0xag09', '0xag09', '0X0'
+];
+
 describe('MonetaryAmount', () => {
 
+    it ('Dummy arguments are legal', () => {
+        expect(() => new MonetaryAmount(0, '0x0000000000000000000000000000000000000000', 0)).to.not.throw();
```

Comment:
Instead of testing what should NOT happen, test what SHOULD happen (i.e. constructor should work and return an instance)

Reply:
My assumption would be that error handling is a feature that should be tested?
Anyway, I have removed the tests.

Reply:
Ok, so I overdid it. Keeping error checking but reducing the scope.